src/edu/washington/escience/myria/operator/CountFilter.java

```diff
@@ -191,8 +187,9 @@ public void value(final int destRow) {
         found = true;
         int oldcount = tupleCounts.getInt(0, destRow);
         if (oldcount < threshold) {
-          tupleCounts.replace(0, destRow, new IntArrayColumn(new int[] {oldcount + 1}, 1), 0);
-          meet = (oldcount + 1 >= threshold);
+          oldcount += 1;
+          tupleCounts.replaceInt(0, destRow, oldcount);
```

Comment:
I believe this won't work unless the new `MutableTupleBuffer.replaceInt()` method is revised to use `destRow % TupleBatch.BATCH_SIZE` as row offset in the column.